src/components/SearchBox/SearchBox.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import './SearchBox.scss';
+
+class SearchBox extends Component {
+  constructor(props) {
+    super();
+
+    this.props = props;
+    this.state = {
+      value: '',
+    };
+
+    this.onInputChange = this.onInputChange.bind(this);
```

Comment:
Once we lose the constructor this function will automatically be bound to the context of the class and this manual bind will become unnecessary.